lib/sqs.js

```diff
@@ -72,13 +154,47 @@ module.exports = ({ account, queuePrefix = '', queueSuffix = '', defaultVisibili
         }
       }
 
-      return SQS.sendMessageAsync({
-        MessageBody: payload,
-        QueueUrl: getQueueURL(queueName),
-        MessageAttributes: attrs
-      });
+      let key = false;
+      const messageStream = compress(payload);
+
+      if (Buffer.byteLength(payload) > TWO_FIFTYSIX_K) {
+        key = `/${prefix}/${Math.floor(Math.random() * shards)}/${uuid()}.json.gz`;
```

Comment:
what's the sharding logic again?  i know it has something to w/ S3 write throttling

Reply:
ive seen referenced to throttling occuring based on differences between the first 6-8 characters, though reading through https://aws.amazon.com/blogs/aws/amazon-s3-performance-tips-tricks-seattle-hiring-event/ i think we may want to be even safer by dropping the `prefix` concept... im sure this is something we'll have to tweak